tests/core/filtering/test_filters_against_many_blocks.py

```diff
@@ -0,0 +1,158 @@
+import pytest
+import random
+
+from eth_utils import (
+    to_tuple,
+)
+
+SEED = 50505050505
+random.seed(SEED)
+
+
+@to_tuple
+def deploy_contracts(web3, contract, wait_for_transaction):
+    for i in range(25):
+        tx_hash = contract.constructor().transact()
+        wait_for_transaction(web3, tx_hash)
+        yield web3.eth.getTransactionReceipt(tx_hash)['contractAddress']
+
+
+def pad_with_transactions(w3):
+    accounts = w3.eth.accounts
+    while True:
+        transact = yield
+        tx_hash = transact()
+        for tx_count in range(random.randint(0, 10)):
+            _from = accounts[random.randint(0, len(accounts) - 1)]
+            _to = accounts[random.randint(0, len(accounts) - 1)]
+            value = 50 + tx_count
+            w3.eth.sendTransaction({'from': _from, 'to': _to, 'value': value})
+        yield tx_hash
```

Comment:
I'm not seeing the benefit of this being a generator. It seems it could be a function you call when you want to pad, like:

```py
def pad_with_transactions(w3):
        accounts = w3.eth.accounts
        for tx_count in range(random.randint(0, 10)):
            _from = accounts[random.randint(0, len(accounts) - 1)]
            _to = accounts[random.randint(0, len(accounts) - 1)]
            value = 50 + tx_count
            w3.eth.sendTransaction({'from': _from, 'to': _to, 'value': value})
```

Then later, instead of invoking the generator and calling `next()`, you call `pad_with_transactions(w3)` again.